### DISCUSSION TITLE
bash: .env: No such file or directory

### DISCUSSION THREAD

**Discussion by khandelwalmoksh** [Cyfrin/foundry-full-course-cu#2724] [Q&A] [answered]
made a .env file, recognized by the .gitignore , still can't find .env file
![Screenshot (24)](https://github.com/user-attachments/assets/d21e0b31-aa94-4f01-95e9-2b3261674472)

**Comment by EngrPips** [comment]
Hello @khandelwalmoksh, You need to have the `.env` file at the root directory.

**Comment by chainflash** [comment]
You don't have the .envy file.
Enter the following command in ubunto to view the root files  :
explorer.exe .

**Reply by khandelwalmoksh** [reply]
Yes, no .env file. Now what to do after the above command
this might help you
![Screenshot (25)](https://github.com/user-attachments/assets/454d5c7f-9163-43e4-8a65-6d4dccb392f5)

the ubuntu 
![Screenshot (27)](https://github.com/user-attachments/assets/969313dc-6f0c-4c31-84a7-1ac259e51a46)

**Comment by EngrPips** [comment]
hmmmm

**Reply by khandelwalmoksh** [reply]
should I manually make a .env file in ubuntu?

**Reply by EngrPips** [reply]
Honestly, I haven't worked with ubuntu before my whole life so it is hard for me to know what works, but try creating it manually and see if that works.

**Reply by khandelwalmoksh** [reply]
I think I should leave this and go ahead. Will try in future. Thanks for helping me

**Reply by EngrPips** [reply]
No worries, Let us know how it plays out whenever you try.

**Reply by khandelwalmoksh** [reply, marked as answer]
Hey @EngrPips , it worked out with the help of CHATGPT. I made .env(manually) in test, now moved to root.
 Move from Test to Root ```mv ./test/.env ./```
and ```source .env```

Or directly ```source ./test/.env```

**Reply by khandelwalmoksh** [reply]
now, I moved forward, changed the codes, It's giving me code error hehe

**Reply by EngrPips** [reply]
What error are you getting now?

**Reply by khandelwalmoksh** [reply]
no worries, debugged it. Smart Contract Security Auditor in making. lfg!!

**Reply by EngrPips** [reply]
I know you will crush these things.
